src/ess/spectroscopy/indirect/workflow.py

```diff
@@ -309,56 +313,73 @@ def get_triplet_events(triplets: Iterable[sc.DataArray]):
     return events
 
 
-def get_sample_events(triplet_events, sample_detector_flight_times):
+def get_sample_events(triplet_events: sc.DataArray, sample_detector_flight_times):
```

Comment:
Type-hint also for `sample_detector_flight_times` and output?

Actually, many functions here are missing type hints. Maybe this will be done in a further cleanup?

Reply:
Most of these functions will disappear or become providers. So yes, I will update the type hints.